tests/test_utilities.py

```diff
@@ -302,13 +302,16 @@ def test_compute_eval_from_predictions_random_choice():
 def check_generate_predictions(use_feature_hashing=False,
                                use_threshold=False,
                                test_on_subset=False,
-                               use_all_labels=False):
-
+                               use_all_labels=False,
+                               string_labels=False):
     # create some simple classification feature sets for training and testing
+    string_label_list = ['a', 'b'] if string_labels else None
+
     train_fs, test_fs = make_classification_data(num_examples=1000,
                                                  num_features=5,
                                                  use_feature_hashing=use_feature_hashing,
-                                                 feature_bins=4)
+                                                 feature_bins=4,
+                                                 string_label_list=string_label_list)
     enable_probability = use_threshold or use_all_labels
```

Comment:
Shouldn't this `enable_probability` also check for `string_labels` being `True`?

Reply:
For this test, setting `enable_probability` to True whenever `string_labels` is True results in a conflict, since when `string_labels` is True we call `learner.predict(test_fs, class_labels=use_class_labels)`, resulting in the class label predictions instead of probabilities. There are cases we would like to test where `string_labels == True` AND `enable_probability == False`, and the only difference wrt string labels when toggling `enable_probaility` occurs in the file header, which is treated by a different test case.

In the `check_generate_predictions_file_header` function, we want to enable probability when labels are strings in order to verify that the output header includes the correct label name(s) ("Probability of <label>", or "id, <label1>, <label2>"). I can add a comment to the test to explain this.